src/content/learn/extracting-state-logic-into-a-reducer.md

```diff
@@ -179,17 +179,17 @@ li {
 
 </Sandpack>
 
-Each of its event handlers calls `setTasks` in order to update the state. As this component grows, so does the amount of state logic sprinkled throughout it. To reduce this complexity and keep all your logic in one easy-to-access place, you can move that state logic into a single function outside your component, **called a "reducer".**
+Setiap _event handler_ pada komponen `TaskApp` akan memangil fungsi `setTask` untuk melakukan pembaharuan _state_. Saat komponen ini bertumbuh semakin besar, logika _state_ akan semakin rumit. Untuk mengurangi kompleksitas dan menyimpan semua logika Anda di satu tempat yang mudah diakses. Anda dapat memindahkan logika _state_ tersebut ke sebuah fungsi di luar komponen Anda, yang **disebut _"reducer"_.**
 
-Reducers are a different way to handle state. You can migrate from `useState` to `useReducer` in three steps:
+_Reducer_ merupakan sebuah alternatif untuk menangani _state_. Anda dapat migrasi dari `useState` ke `useReducer` dalam tiga langkah:
 
-1. **Move** from setting state to dispatching actions.
-2. **Write** a reducer function.
-3. **Use** the reducer from your component.
+1. **Pindah** dari memanggil fungsi penetap _state_ menjadi aksi _dispatch_.
+2. **Tulis** fungsi _reducer_.
+3. **Gunakan** _reducer_ pada komponen Anda.
 
-### Step 1: Move from setting state to dispatching actions {/*step-1-move-from-setting-state-to-dispatching-actions*/}
+### Langkah 1: Pindah dari memanggil fungsi penetap state menjadi aksi dispatch {/*step-1-move-from-setting-state-to-dispatching-actions*/}
```

Comment:
Lihat komentar sebelumnya, samain biar konsisten.

Reply:
```suggestion
### Langkah 1: Pindah dari menyetel state menjadi men-dispatch action {/*step-1-move-from-setting-state-to-dispatching-actions*/}
```

Idem.

Reply:
Ini untuk italic:

```suggestion
### Langkah 1: Pindah dari menyetel *state* menjadi men-*dispatch* *action* {/*step-1-move-from-setting-state-to-dispatching-actions*/}
```

Reply:
@mhaidarhanif No need to italicise for headings.